controller_manager:
  ros__parameters:
    update_rate: 100

    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_controller:
      type: position_controllers/JointGroupPositionController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

arm_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
    command_interfaces:
      - position
    state_interfaces:
      - position
    allow_partial_joints_goal: false
    open_loop_control: true
    interpolate_from_desired_state: true
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    stop_trajectory_duration: 0.25
    constraints:
      goal_time: 5.0
      stopped_velocity_tolerance: 0.2
      joint_1:
        trajectory: 2.50
        goal: 2.50
      joint_2:
        trajectory: 2.50
        goal: 2.50
      joint_3:
        trajectory: 2.50
        goal: 2.50
      joint_4:
        trajectory: 2.50
        goal: 2.50

gripper_controller:
  ros__parameters:
    joints:
      - joint_gripper
    interface_name: position
